### DISCUSSION TITLE
Product availability based on what product(s) have been added to a cart

### DISCUSSION THREAD

**Discussion by augustin** [pretix/pretix#2575] [Ideas] [answered]
I have an event at which, e.g., children under a certain age can be registered for free, but only after a certain number of paying registrants. This is presently accomplished by having the "child" product be an add-on product to another one, which has a "minimum per order" set.

However, having to enter how many children are being registered in the "add-on" stage has proven confusing to users, as well as the fact that there are as many boxes for the "child" add-on as there are products added to the cart.

It would be much more ideal if a different solution could be found. For instance, a way to set "minimum per order" of product *variations* so that one variation could be for adults and another for children; or a way to enable products only if another product(s) have been added to the cart.

**Comment by raphaelm** [comment, marked as answer]
While I understand the need, I'm pretty sure this won't happen. As we currently notice during implementing #2510, every kind of inter-dependency between cart positions that goes beyond the addon-parent relationship makes everything **really complicated**. There's just so many edge cases to think of, I believe your issue does not warrant this level of complexity – sorry :(

**Reply by augustin** [reply]
Not even possible to have a "minimum per order" per product variant (i.e. requiring that if any of the product are added to the order, a minimum number of that variant must be added)?

**Reply by augustin** [reply]
Already there is an option for "product is invisible unless a voucher is redeemed," it would be good to have such a similar option but for having some other product in the cart.

Or, even just having an option that add-ons be chosen along with other products before checkout; that would also make things less confusing.

**Reply by augustin** [reply]
If none of this is possible, then at least the option to add more explanatory text to the "Add-ons" stage so users are less confused would be helpful...

**Reply by raphaelm** [reply]
> Not even possible to have a "minimum per order" per product variant (i.e. requiring that if any of the product are added to the order, a minimum number of that variant must be added)?

No, I don't think we'll add this.

> Already there is an option for "product is invisible unless a voucher is redeemed," it would be good to have such a similar option but for having some other product in the cart.

The latter much much more complex than the former. Just to give an example, if we'd build this, what happens if you add 2 adults and 3 childs to the cart, and then remove the adults again? We'd need to keep track of all these inter-dependencies to make sure you can't cheat the system. 

> Or, even just having an option that add-ons be chosen along with other products before checkout; that would also make things less confusing.

This kinda goes against the entire idea of addons and I think would be even more confusing for 95% of use cases.

> If none of this is possible, then at least the option to add more explanatory text to the "Add-ons" stage so users are less confused would be helpful...

That would make sense.


There is of course one additional option here: You can build a plugin that implements the validate_cart signal to perform additional logic to check whether a cart contains an allowed configuration of products in whichever way you like.
